mono/mini/interp/interp.c

```diff
@@ -2071,6 +2067,18 @@ do_icall (InterpFrame *frame, MonoMethodSignature *sig, int op, stackval *sp, gp
 	if (sig)
 		stackval_from_data (sig->ret, &sp [-1], (char*) &sp [-1].data.p, sig->pinvoke);
 
+	return sp;
+}
+
+/* MONO_NO_OPTIMIATION is needed due to usage of INTERP_PUSH_LMF_WITH_CTX. */
+static MONO_NO_OPTIMIZATION MONO_NEVER_INLINE stackval *
+do_icall_wrapper (InterpFrame *frame, MonoMethodSignature *sig, int op, stackval *sp, gpointer ptr)
+{
+	MonoLMFExt ext;
+	INTERP_PUSH_LMF_WITH_CTX (frame, ext, &&exit_icall);
+
+	sp = do_icall (frame, sig, op, sp, ptr);
+
```

Comment:
I'm a bit afraid that could break the exception handling stuff on tvOS.  do we really need that?

Reply:
I don't see why it would break, and if it does, it means the whole mechanism is unreliable, and would need fixing.

Reply:
unfortunately it _is_ fragile.  we can still merge it, it will be easy to undo